oak_enclave_runtime_support/src/heap.rs

```diff
@@ -79,14 +79,23 @@ impl GrowableHeap {
         }
     }
 
+    /// # Safety
+    ///
+    /// Upheld by caller.
     pub unsafe fn init(&mut self) {}
 
+    #[allow(clippy::result_unit_err)]
```

Comment:
@andrisaar likewise this seems like the error could be the currently logged static string. It seems surprising that we log here, but also register an alloc error in enclave apps.